### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by laoneo**
### Summary of Changes
Updates the dependencies for the CMS. This composer.json and package.json files are reflecting now the installed versions of the dependencies, like we can easily see what has changed.

To update the PHP dependencies I did run the following commands:
`composer bump`
`composer update`

To update the asset (Javascript, CSS) dependencies I did run the following commands
`npm update --save`
`npm run update`

### Testing Instructions
Browse around in the back end.

### Actual result BEFORE applying this Pull Request
All works.

### Expected result AFTER applying this Pull Request
All works.

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [x] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [x] No documentation changes for manual.joomla.org needed

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -39,74 +39,74 @@
...
+    "stylelint": "^14.16.1",
     "stylelint-config-standard": "^24.0.0",
     "stylelint-order": "^5.0.0",
-    "stylelint-scss": "^4.0.1",
-    "terser": "^5.7.0"
+    "stylelint-scss": "^4.4.0",
+    "terser": "^5.16.3"
   },
   "optionalDependencies": {
     "sass-embedded-linux-x64": "^1.57.1"
```
Comment (dgrammatiko) [inline package.json]: Change the version here to match the sass-embedded a few lines before

Comment (laoneo) [reply package.json]: This doesn't help. Does it work for you?

Comment (dgrammatiko) [reply package.json]: Let me check

Comment (dgrammatiko) [reply package.json]: Actually the sass pkg is broken: https://github.com/sass/embedded-host-node/issues/205

Comment (laoneo) [reply package.json]: This issue was not there as I was searching for it yesterday :sweat_smile: . So I guess we have to hardcode the old version till it is resolved.

Comment (dgrammatiko) [reply package.json]: @laoneo you can use version `1.58.3`

**Comment by dgrammatiko** [comment]
@laoneo couple of things here:
- you should remove the `optionalDependencies` from the package.json, my bad introducing it
- You should run `npm run update` and commit the xmls of tinyMCE and codemirror and also the error pages

**Comment by laoneo** [comment]
`npm run update` has no effect, no files are changed

**Comment by dgrammatiko** [comment]
> npm run update has no effect, no files are changed

then maybe `npm i`? The XMLs and the error pages need to be updated

**Comment by laoneo** [comment]
Had a look on the XML files and they do have the correct version already. What for error pages do you mean?

**Comment by dgrammatiko** [comment]
> What for error pages do you mean?

These: templates/system/build_incomplete.html, templates/system/fatal-error.html, templates/system/incompatible.html

**Comment by brianteeman** [comment]
Not going to pretend to understand npm versioning but I would expect at a minimum that after applying this pr and then running npm i to have the latest version that follows the versioning defined in package.json but the package-lock.json is not providing those versions

`npm outdated` 

PS C:\htdocs\j42> npm outdated
```
Package                        Current   Wanted  Latest  Location                                    Depended by
@babel/core                    7.20.12   7.21.0  7.21.0  node_modules/@babel/core                    j42
@dgrammatiko/compress            1.0.4    1.0.4   2.0.0  node_modules/@dgrammatiko/compress          j42
@fortawesome/fontawesome-free   5.15.4   5.15.4   6.3.0  node_modules/@fortawesome/fontawesome-free  j42
@rollup/plugin-babel             5.3.1    5.3.1   6.0.3  node_modules/@rollup/plugin-babel           j42
@rollup/plugin-commonjs         21.1.0   21.1.0  24.0.1  node_modules/@rollup/plugin-commonjs        j42
@rollup/plugin-node-resolve     13.3.0   13.3.0  15.0.1  node_modules/@rollup/plugin-node-resolve    j42
@rollup/plugin-replace           3.1.0    3.1.0   5.0.2  node_modules/@rollup/plugin-replace         j42
accessibility                   3.0.17   3.0.17   4.5.6  node_modules/accessibility                  j42
choices.js                       9.1.0    9.1.0  10.2.0  node_modules/choices.js                     j42
codemirror                     5.65.11  5.65.12   6.0.1  node_modules/codemirror                     j42
commander                        8.3.0    8.3.0  10.0.0  node_modules/commander                      j42
cypress                        10.11.0  10.11.0  12.7.0  node_modules/cypress                        j42
eslint-plugin-vue                8.7.1    8.7.1   9.9.0  node_modules/eslint-plugin-vue              j42
fs-extra                        10.1.0   10.1.0  11.1.0  node_modules/fs-extra                       j42
ini                              2.0.0    2.0.0   3.0.1  node_modules/ini                            j42
jasmine-core                    3.99.1   3.99.1   4.5.0  node_modules/jasmine-core                   j42
jquery-migrate                   3.4.0    3.4.1   3.4.1  node_modules/jquery-migrate                 j42
mediaelement                     5.1.1    5.1.1   6.0.1  node_modules/mediaelement                   j42
rimraf                           3.0.2    3.0.2   4.1.2  node_modules/rimraf                         j42
rollup                          2.79.1   2.79.1  3.17.3  node_modules/rollup                         j42
rtlcss                           3.5.0    3.5.0   4.0.0  node_modules/rtlcss                         j42
stylelint                      14.16.1  14.16.1  15.2.0  node_modules/stylelint                      j42
stylelint-config-standard       24.0.0   24.0.0  30.0.1  node_modules/stylelint-config-standard      j42
stylelint-order                  5.0.0    5.0.0   6.0.2  node_modules/stylelint-order                j42
terser                          5.16.4   5.16.5  5.16.5  node_modules/terser                         j42
tinymce                         5.10.7   5.10.7   6.3.2  node_modules/tinymce                        j42
vue                             3.2.45   3.2.45  3.2.47  node_modules/vue                            j42
vue-focus-lock                   2.0.2    2.0.5   2.0.5  node_modules/vue-focus-lock                 j42
```

**Comment by laoneo** [comment]
@brianteeman most of these are major updates which should be done in joomla 5 branch.

**Comment by brianteeman** [comment]
Updated the list to only sure minor updates
```

Package                        Current   Wanted  Latest
@babel/core                    7.20.12   7.21.0  7.21.0
codemirror                     5.65.11  5.65.12   6.0.1
jquery-migrate                   3.4.0    3.4.1   3.4.1
terser                          5.16.4   5.16.5  5.16.5
vue-focus-lock                   2.0.2    2.0.5   2.0.5
```

**Comment by laoneo** [comment]
Can you check with the latest commits? Looks like they are updated since I did it last time.

**Comment by brianteeman** [comment]
I have tested this item :white_check_mark: successfully on b5ef5311b02eec6a43e0021455b6547f3db3e735<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39875">issues.joomla.org/tracker/joomla-cms/39875</a>.</sub>

**Comment by dgrammatiko** [comment]
I have tested this item :white_check_mark: successfully on b5ef5311b02eec6a43e0021455b6547f3db3e735<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39875">issues.joomla.org/tracker/joomla-cms/39875</a>.</sub>

**Comment by richard67** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39875">issues.joomla.org/tracker/joomla-cms/39875</a>.</sub>

**Comment by obuisard** [comment]
Thank you Allon @laoneo

**Comment by laoneo** [comment]
Coo, thank you @obuisard !!

**Pull request merged**